autoendpoint/src/headers/vapid.rs

```diff
@@ -149,26 +149,24 @@ impl VapidHeader {
     /// WARNING: THIS FUNCTION DOES NOT VALIDATE THE VAPID HEADER AND SHOULD
     /// ONLY BE USED FOR LOGGING AND METRIC REPORTING FUNCTIONS.
     /// Proper validation should be done by [crate::extractors::subscription::validate_vapid_jwt()]
-    pub fn insecure_sub(&self) -> Result<String, VapidError> {
+    pub fn insecure_sub(&self) -> Result<Option<String>, VapidError> {
         // This parses the VAPID header string
         let data = VapidClaims::try_from(self.clone()).inspect_err(|e| {
             warn!("🔐 Vapid: {:?} {:?}", e, &self.token);
         })?;
 
-        if let Some(sub) = data.sub {
-            if !sub.starts_with("mailto:") && !sub.starts_with("https://") {
-                info!("🔐 Vapid: Bad Format {:?}", sub);
-                return Err(VapidError::SubBadFormat);
+        let sub = match data.sub {
+            Some(sub) => {
+                if !sub.starts_with("mailto:") && !sub.starts_with("https://") {
+                    info!("🔐 Vapid: Bad Format {:?}", sub);
+                    return Err(VapidError::SubBadFormat);
+                }
+                Some(sub)
```

Comment:
let's do a `let else` --

I'll note [RFC8292](https://www.rfc-editor.org/rfc/rfc8292) says sub **SHOULD** be a mailto/https, so maybe we shouldn't even consider other schemes as errors (but it's only for our logging/debugging purposes anyway). However I'd argue the empty check should be kept if we're continuing to consider other schemes as errors.

```suggestion
        let Some(sub) = data.sub else {
            return Ok(None);
        };
        if !sub.starts_with("mailto:") && !sub.starts_with("https://") {
            info!("🔐 Vapid: Bad Format {sub:?}");
            return Err(VapidError::SubBadFormat);
        }
        if sub.is_empty() {
            info!("🔐 Empty Vapid sub");
            return Err(VapidError::SubEmpty);
        }
        info!("🔐 Vapid: sub: {sub}");
        Ok(Some(sub))
```